pyre/extra_tests/snippets/stdlib_itertools.py

```
@@ -62,6 +63,15 @@
 with assert_raises(StopIteration):
     next(x)
 
+# Python 3.14 deliberately omits PyPy's historical chain pickle state
+# methods.  The inherited object reducer rejects this native iterator, and no
+# instance-only fallback may make __setstate__ appear outside the TypeDef.
+x = chain([1], [2])
+with assert_raises(TypeError):
+    pickle.dumps(x)
+with assert_raises(AttributeError):
+    x.__setstate__
```

Ruff (0.16.2)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)